src/neo/SmartContract/Native/Tokens/NeoToken.Economic.cs

```diff
@@ -0,0 +1,178 @@
+        {
+            var endRewardItem = snapshot.Storages.TryGet(CreateStorageKey(Prefix_HolderRewardPerBlock, uint.MaxValue - end - 1));
+            var startRewardItem = snapshot.Storages.TryGet(CreateStorageKey(Prefix_HolderRewardPerBlock, uint.MaxValue - start - 1));
+            BigInteger startReward = startRewardItem is null ? 0 : new BigInteger(startRewardItem.Value);
+            return value * (new BigInteger(endRewardItem.Value) - startReward) / TotalAmount;
+        }
+
+        [ContractMethod(0_03000000, CallFlags.AllowStates)]
+        public BigInteger UnclaimedGas(StoreView snapshot, UInt160 account, uint end)
+        {
+            StorageItem storage = snapshot.Storages.TryGet(CreateAccountKey(account));
+            if (storage is null) return BigInteger.Zero;
+            NeoAccountState state = storage.GetInteroperable<NeoAccountState>();
+            return CalculateBonus(snapshot, state.VoteTo, state.Balance, state.BalanceHeight, end);
+        }
+
+        private void DistributeGasForCommittee(ApplicationEngine engine)
+        {
+            var gasPerBlock = GetGasPerBlock(engine.Snapshot);
+            (ECPoint, BigInteger)[] committeeVotes = GetCommitteeVotes(engine.Snapshot);
+            int validatorNumber = GetValidators(engine.Snapshot).Length;
+            int totalRewardRatio = GetTotalRewardRatio(engine.Snapshot);
+            BigInteger holderRewardPerBlock = gasPerBlock * GetNeoHoldersRewardRatio(engine.Snapshot) / totalRewardRatio;
+            BigInteger committeeRewardPerBlock = gasPerBlock * GetCommitteeRewardRatio(engine.Snapshot) / totalRewardRatio / committeeVotes.Length;
+            BigInteger voterRewardPerBlock = gasPerBlock * GetVotersRewardRatio(engine.Snapshot) / totalRewardRatio / (committeeVotes.Length + validatorNumber);
+
+            // Keep track of incremental gains of neo holders
+
+            var index = engine.Snapshot.PersistingBlock.Index;
+            var holderRewards = holderRewardPerBlock;
+            var holderRewardKey = CreateStorageKey(Prefix_HolderRewardPerBlock, uint.MaxValue - index - 1);
+            var holderBorderKey = CreateStorageKey(Prefix_HolderRewardPerBlock, uint.MaxValue);
+            var enumerator = engine.Snapshot.Storages.FindRange(holderRewardKey, holderBorderKey).GetEnumerator();
+            if (enumerator.MoveNext())
+                holderRewards += new BigInteger(enumerator.Current.Value.Value);
+            engine.Snapshot.Storages.Add(holderRewardKey, new StorageItem() { Value = holderRewards.ToByteArray() });
+
+            for (var i = 0; i < committeeVotes.Length; i++)
+            {
+                // Keep track of incremental gains of committee voters
+
+                UInt160 committeeAddr = Contract.CreateSignatureContract(committeeVotes[i].Item1).ScriptHash;
+                BigInteger voterRewardPerCommittee = (i < validatorNumber ? 2 : 1) * voterRewardPerBlock * 10000L / committeeVotes[i].Item2;
```

Comment:
Is it distributed equally or the number of votes of each committee will affect? If true  maybe we need to update description.

Reply:
The number of votes of each committee will affect, I'll update description.

Reply:
Good, @Tommo-L,

What I am thinking, @vang1ong7ang, is that people will change vote quick in order to maximize their gains, no?

Reply:
We are considering a multiplier `2` if you vote for CN instead of non-CN in the Committee.
If you vote for someone out of the Committee then you will soon change as well, right?

Reply:
Did you update the description, @Tommo-L ?

Reply:
> We are considering a multiplier 2 if you vote for CN instead of non-CN in the Committee.

Yes

> If you vote for someone out of the Committee then you will soon change as well, right?

I don't think so, it'll eventually reach a dynamic balance.

Reply:
> Did you update the description, @Tommo-L ?

Done.